File: eHour-persistence/src/main/java/net/rrm/ehour/persistence/dao/AbstractAnnotationDaoHibernate4Impl.java
package net.rrm.ehour.persistence.dao;

import org.hibernate.Session;
import org.hibernate.SessionFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Repository;

@Repository
public class AbstractAnnotationDaoHibernate4Impl {
    @Autowired
    private SessionFactory sessionFactory;

    protected Session getSession() {
        return sessionFactory.getCurrentSession();
    }
}

File: eHour-persistence/src/main/scala/net/rrm/ehour/persistence/dao/AbstractAnnotationDaoHibernate4Impl.scala
package net.rrm.ehour.persistence.dao

import org.hibernate.SessionFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Repository

@Repository
class AbstractAnnotationDaoHibernate4Impl {
  @Autowired
  var sessionFactory: SessionFactory = _

  def getSession = sessionFactory.getCurrentSession
}
